draft-ietf-tls-tls13.md

```diff
@@ -2782,9 +2789,11 @@ can behave in one of two ways:
   to accept only a subset of the early data messages.
 
 [[OPEN ISSUE: are the rules below correct? https://github.com/tlswg/tls13-spec/issues/451]]
-Prior to accepting the "early_data" extension, the server MUST
-validate that the session ticket parameters are consistent with its
-current configuration. It MUST also validate that the extensions
+Prior to accepting the "early_data" extension, the server MUST check the
+first PSK identity (see {{pre-shared-key-extension}}) corresponds to a
+valid ticket and validate that the session parameters are consistent
+with its current configuration.
```

Comment:
I would state this more simply: If the server selects a PSK identity other than the first, it MUST reject early data.  

That's stronger than what you have, but I think that it also suggests an order of operations that is inherently safer: pick your parameters and only then work out whether to take this 0-RTT stuff.  I know that some implementations will want to maximize how often 0-RTT is used, but that isn't necessarily the best choice security-wise.